### ISSUE TITLE
False positive with pylxd module

### ISSUE THREAD

**Issue by pcgeek86** [pylint-dev/pylint#8965]
### Bug description

When attempting to call the `.all()` method on `client.virtual_machines`, pylint is throwing as "has no member" error message.

However, the code actually runs correctly.

```python
from pylxd import Client

client = Client()
print(client.instances)

print(client.virtual_machines)
print(dir(client.virtual_machines))
print(client.virtual_machines.all())
```

<img width="1083" alt="image" src="https://github.com/pylint-dev/pylint/assets/466713/787df327-282f-4d90-8d06-a13aee6a952e">

<img width="1501" alt="image" src="https://github.com/pylint-dev/pylint/assets/466713/02e7b7a2-0e2f-4af8-95b9-7a36d29ddf1e">

### Command used

```shell
pylint lxd_automation.py
```

### Pylint output

```shell
************* Module lxd-automation
lxd-automation.py:1:0: C0103: Module name "lxd-automation" doesn't conform to snake_case naming style (invalid-name)
lxd-automation.py:11:6: E1101: Instance of 'VirtualMachineManager' has no 'all' member (no-member)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```
```


### Expected behavior

No error should be thrown in this case.

### Pylint version

```shell
pylint 2.17.5
astroid 2.15.6
Python 3.10.12 (main, Jun 11 2023, 05:26:28) [GCC 11.4.0]
```
```


### OS / Environment

Ubuntu Server 22.04 LTS

### Additional dependencies

n/a
